### PR TITLE
Background pass

### PR DESCRIPTION
**Pull request by arose**
TODOs
- needs more testing
- rethink ``sceneRadiusFactor``, maybe also add an option for a fixed radius
- find better skybox example

DONE
- Add ``fov`` (Field of View) Canvas3D parameter
- Add ``sceneRadiusFactor`` Canvas3D parameter
- Add background pass (skybox, image, horizontal/radial gradient)
    - Set simple-settings presets via ``PluginConfig.Background.Styles``
    - Example presets in new backgrounds extension
    - Load skybox/image from URL or File (saved in session)
    - Opacity, saturation, lightness controls for skybox/image
    - Coverage (viewport or canvas) controls for image/gradient
- [Breaking] ``AssetManager`` needs to be passed to various graphics related classes

### CODE REVIEW

**Review by dsehnal — COMMENTED** [review]
This looks super cool. Worked locally for me without problems.

File: src/mol-canvas3d/passes/postprocessing.ts [hunk 1]
```diff
@@ -171,15 +173,26 @@ function getBlurKernel(kernelSize: number): number[] {
     return kernel;
 }
 
-function getSamples(vectorSamples: Vec3[], nSamples: number): number[] {
+const RandomHemisphereVector: Vec3[] = [];
+for (let i = 0; i < 256; i++) {
+    const v = Vec3();
+    v[0] = Math.random() * 2.0 - 1.0;
+    v[1] = Math.random() * 2.0 - 1.0;
+    v[2] = Math.random();
```
[the remark this answers is not in the document]
Comment (arose) [reply src/mol-canvas3d/passes/postprocessing.ts:181]: Maybe. This pattern is quite commonly used for SSAO directions. It's not new here, just moved it, didn't came up as an issue visually before.

File: src/extensions/backgrounds/typings.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright (c) 2022 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ */
+
+declare module '*.jpg' {
+    const value: any;
```
[the remark this answers is not in the document]
Comment (arose) [reply src/extensions/backgrounds/typings.d.ts:8]: yes, thanks

File: src/mol-canvas3d/passes/background.ts [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+            this.skybox = undefined;
+        }
+    }
+
+    private updateSkybox(camera: ICamera, props: SkyboxProps, onload?: (changed: boolean) => void) {
+        const tf = this.skybox?.props.faces;
+        const f = props.faces.params;
+        if (!f.nx || !f.ny || !f.nz || !f.px || !f.py || !f.pz) {
+            this.clearSkybox();
+            if (onload) onload(false);
```
[the remark this answers is not in the document]
Comment (arose) [reply src/mol-canvas3d/passes/background.ts:140]: yes, it means changed asynchronously

File: src/mol-canvas3d/passes/image.ts [hunk 4]
```diff
@@ -63,6 +64,14 @@ export class ImagePass {
         this.setSize(1024, 768);
     }
 
+    async updateBackground() {
```
[the remark this answers is not in the document]
Comment (arose) [reply src/mol-canvas3d/passes/image.ts:67]: :)

File: src/mol-canvas3d/passes/draw.ts [hunk 5]
```diff
@@ -235,7 +235,7 @@ export class DrawPass {
         }
     }
 
-    private _render(renderer: Renderer, camera: ICamera, scene: Scene, helper: Helper, toDrawingBuffer: boolean, props: Props) {
+    private _render(renderer: Renderer, camera: ICamera, scene: Scene, helper: Helper, toDrawingBuffer: boolean, transparentBackground: boolean, props: Props) {
```
[the remark this answers is not in the document]
Comment (arose) [reply src/mol-canvas3d/passes/draw.ts:238]: no comment... one day

File: src/mol-gl/webgl/texture.ts [hunk 6]
```diff
@@ -423,6 +424,118 @@ export function loadImageTexture(src: string, cell: ValueCell<Texture>, texture:
...
+                if (mipmaps) {
+                    gl.generateMipmap(target);
+                    gl.texParameteri(target, gl.TEXTURE_MIN_FILTER, gl.LINEAR_MIPMAP_LINEAR);
+                } else {
+                    gl.texParameteri(target, gl.TEXTURE_MIN_FILTER, filter);
+                }
+                gl.texParameteri(target, gl.TEXTURE_MAG_FILTER, filter);
+                if (onload) onload();
+            }
+        });
```
[the remark this answers is not in the document]
Comment (arose) [reply src/mol-gl/webgl/texture.ts:495]: yes, good point, will handle errors

**Pull request merged**